pkg/plugins/manager/manager.go

```diff
@@ -228,7 +228,7 @@ func (m *PluginManager) Renderer() *plugins.Plugin {
 func (m *PluginManager) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	plugin := m.Plugin(req.PluginContext.PluginID)
 	if plugin == nil {
-		return &backend.QueryDataResponse{}, nil
+		return &backend.QueryDataResponse{}, backendplugin.ErrPluginNotRegistered
```

Comment:
does this cover the case where the plugin exists, but the executable it expects to be there is not found?  That would save a lot of headaches if the error were more clear

Reply:
No, that particular case shouldn't happen now as a plugin won't get loaded if we recognize a signature "modification" under any circumstance - that means either contents on the manifest have been fudged, or some file that's listed in the manifest. So in the case of a missing executable, the plugin won't be part of the manager at all and you'll get a server log for that. You will also see a `modified` tab on the plugin card within the catalog.

We're planning to improve this entire part of plugins soon as we look into more formal plugin state management